README
vThemes v1.6.0 Plugin README

vThemes is a Visual Theme Switcher and Browser for Geeklog v2.2.1 or higher.

For detailed installation and usage documentation, please refer to the /docs
directory.

For Theme Authors who want to support vThemes:

vThemes will look for the following file in the theme directory:
  - themefoldername.png or themefoldername.jpg - This is a screen shot 
	of the theme (usually 800 pixels wide or more)

All other information about the theme including name, version, description, 
author, etc is retrieved from the themes functions.php and theme_config_foo 
function. Take a look at the themes included with Geeklog for examples.


docs/ChangeLog
vThemes Plugin ChangeLog

v1.6.1 (Released 2020-04-23)
  - [Info] Requires Geeklog 2.2.1 or higher
  - [Info] Requires PHP v5.6.4 or higher (including PHP v7)
  - [Security] Removed timthumb due to security concerns with image resizing php script
  - [Improvement] Now uses information directly from theme config so removed .ini file requirement
  - [Feature] Added multiple theme template support
  - [Feature] More information is displayed in Visual Switcher about theme (if available)
  - [Feature] Actual theme name used now in Visual Switcher Block
  - [Bug] In Visual Switcher fixed multiple blank rows at bottom
  - [Bug] Other misc. bug fixes
  - [Bug] Removed broken jQuery lightbox code. Denim theme will use uikit lightbox
  
v1.6.0 
  - ?
  
v1.5.2
  - [feature] Need jQuery plugin to display preview in lightbox

v1.5.1.1
  - imp - Install script. Clean a hello :) Attribution $pi_version and change $gl_version

v1.5.1
  - imp - Theme
  - fix - Number of themes in the block theme's switcher.
  - new - Description and disable download in theme.ini
  - new - Online config

v1.0.2
  - fix - Improved HTML to validate as HTML 4.01 Strict
  - fix - Fixed issue in setting cookie after theme selection

v1.0.1
  - fix - If allow_themes is 0 in Geeklog config.php, no themes would appear in browser.
  - fix - sorting was too specific, always placing uppercase items on top.
    now use natural order sort.
  - fix - now sort the vThemes block array

v1.0.0 - Initial Public Release


autoinstall.php
<?php

/* Reminder: always indent with 4 spaces (no tabs). */
// +---------------------------------------------------------------------------+
// | vthemes Plugin 1.6                                                       |
// +---------------------------------------------------------------------------+
// | autoinstall.php                                                           |
// |                                                                           |
// | This file provides helper functions for the automatic plugin install.     |
// +---------------------------------------------------------------------------+
// | Copyright (C) 2008-2010 by the following authors:                         |
// |                                                                           |
// | Authors: Dirk Haun         - dirk AT haun-online DOT de                   |
// +---------------------------------------------------------------------------+
// |                                                                           |
// | This program is free software; you can redistribute it and/or             |
// | modify it under the terms of the GNU General Public License               |
// | as published by the Free Software Foundation; either version 2            |
// | of the License, or (at your option) any later version.                    |
// |                                                                           |
// | This program is distributed in the hope that it will be useful,           |
// | but WITHOUT ANY WARRANTY; without even the implied warranty of            |
// | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the             |
// | GNU General Public License for more details.                              |
// |                                                                           |
// | You should have received a copy of the GNU General Public License         |
// | along with this program; if not, write to the Free Software Foundation,   |
// | Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.           |
// |                                                                           |
// +---------------------------------------------------------------------------+

/**
* vthemes plugin automatic plugin install
*
* @package vthemes
*/

function plugin_autoinstall_vthemes($pi_name)
{
    $pi_name         = 'vthemes';
    $pi_display_name = 'vthemes';
    $pi_admin        = $pi_display_name . ' Admin';

    $info = array(
        'pi_name'         => $pi_name,
        'pi_display_name' => $pi_display_name,
        'pi_version'      => '1.6.1',
        'pi_gl_version'   => '2.2.1',
        'pi_homepage'     => 'https://github.com/Geeklog-Plugins/vthemes'
    );

    $groups = array(
        $pi_admin => 'Has full access to ' . $pi_display_name . ' features'
    );

    $features = array(
        $pi_name . '.admin' => 'Full access to ' . $pi_display_name . ' plugin'
    );

    $mappings = array(
        $pi_name . '.admin' => array($pi_admin)
    );

    $tables = array(
    );

    $inst_parms = array(
        'info'      => $info,
        'groups'    => $groups,
        'features'  => $features,
        'mappings'  => $mappings,
        'tables'    => $tables
    );

    return $inst_parms;
}

function plugin_load_configuration_vthemes($pi_name)
{
    global $_CONF;

    $base_path = $_CONF['path'] . 'plugins/' . $pi_name . '/';

    require_once $_CONF['path_system'] . 'classes/config.class.php';
    require_once $base_path . 'install_defaults.php';

    return plugin_initconfig_vthemes();
}

function plugin_postinstall_vthemes($pi_name)
{
    return true;
}

function plugin_compatible_with_this_version_vthemes($pi_name)
{
    if (!function_exists('SEC_createToken')) {
        return false;
    }

    return true;
}

?>


install_updates.php
<?php

function vthemes_update_ConfValues_1_5_3()
{
    global $_VT_CONF, $_VT_DEFAULT, $_GROUPS, $_TABLES;

    require_once $_CONF['path_system'] . 'classes/config.class.php';

    $c = config::get_instance();

    require_once $_CONF['path'] . 'plugins/vthemes/install_defaults.php';

    $c->add('sg_main', NULL, 'subgroup', 0, 0, NULL, 0, true, 'vthemes', 0);
    $c->add('tab_main', NULL, 'tab', 0, 0, NULL, 0, true, 'vthemes', 0);
    $c->add('fs_access', NULL, 'fieldset', 0, 0, NULL, 0, true, 'vthemes', 0);
    $c->add('vthemesloginrequired', $_VT_DEFAULT['vthemesloginrequired'],
            'select', 0, 0, 0, 10, true, 'vthemes', 0);
    $c->add('hidevthemesmenu', $_VT_DEFAULT['hidevthemesmenu'], 'select',
            0, 1, 0, 20, true, 'vthemes', 0);
    $c->add('fs_visual', NULL, 'fieldset', 0, 1, NULL, 0, true, 'vthemes', 0);
    $c->add('max_width_preview', $_VT_DEFAULT['max_width_preview'], 'text',
            0, 0, 0, 30, true, 'vthemes', 0);
    $c->add('columns', $_VT_DEFAULT['columns'], 'text',
            0, 0, 0, 40, true, 'vthemes', 0);
    $c->add('rows', $_VT_DEFAULT['rows'], 'text',
            0, 0, 0, 50, true, 'vthemes', 0);
    $c->add('disable_use_theme', $_VT_DEFAULT['disable_use_theme'], 'select',
            0, 1, 0, 60, true, 'vthemes', 0);
    $c->add('disable_download', $_VT_DEFAULT['disable_download'], 'select',
            0, 1, 0, 70, true, 'vthemes', 0);
    $c->add('leftblocks', $_VT_DEFAULT['leftblocks'], 'select',
            0, 1, 0, 100, true, 'vthemes', 0);
    
    //Block tab
    $c->add('tab_vthemes_block', NULL, 'tab', 0, 20, NULL, 0, true, 'vthemes', 10);
    $c->add('fs_block_settings', NULL, 'fieldset', 0, 10, NULL, 0, true, 'vthemes', 10);
    $c->add('block_enable', $_VT_DEFAULT['block_enable'], 'select', 
            0, 10, 0, 10, true, 'vthemes', 10);
    $c->add('block_isleft', $_VT_DEFAULT['block_isleft'], 'select', 
            0, 10, 0, 20, true, 'vthemes', 10);
    $c->add('block_order', $_VT_DEFAULT['block_order'], 'text',
            0, 10, 0, 30, true, 'vthemes', 10);
    $c->add('block_topic_option', $_VT_DEFAULT['block_topic_option'],'select',
            0, 10, 15, 40, true, 'vthemes', 10);  
    $c->add('block_topic', $_VT_DEFAULT['block_topic'], '%select',
            0, 10, NULL, 50, true, 'vthemes', 10);

    $c->add('fs_block_permissions', NULL, 'fieldset', 0, 20, NULL, 0, true, 'vthemes', 10);
    $new_group_id = 0;
    if (isset($_GROUPS['vthemes Admin'])) {
        $new_group_id = $_GROUPS['vthemes Admin'];
    } else {
        $new_group_id = DB_getItem($_TABLES['groups'], 'grp_id', "grp_name = 'vthemes Admin'");
        if ($new_group_id == 0) {
            if (isset($_GROUPS['Root'])) {
                $new_group_id = $_GROUPS['Root'];
            } else {
                $new_group_id = DB_getItem($_TABLES['groups'], 'grp_id', "grp_name = 'Root'");
            }
        }
    }         
    $c->add('block_group_id', $new_group_id,'select',
            0, 20, NULL, 10, TRUE, 'vthemes', 10);        
    $c->add('block_permissions', $_VT_DEFAULT['block_permissions'], '@select', 
            0, 20, 14, 20, true, 'vthemes', 10);

    return true;
}

?>

install_defaults.php
<?php

// Reminder: always indent with 4 spaces (no tabs).
// +---------------------------------------------------------------------------+
// | vthemes Plugin 1.5.1                                                          |
// +---------------------------------------------------------------------------+
// | install_defaults.php                                                      |
// |                                                                           |
// | Initial Installation Defaults used when loading the online configuration  |
// | records. These settings are only used during the initial installation     |
// | and not referenced any more once the plugin is installed.                 |
// +---------------------------------------------------------------------------+
// | Copyright (C) 2007 by the following authors:                              |
// |                                                                           |
// | Author: Mark R. Evans - mevans@ecsnet.com                                 |
// +---------------------------------------------------------------------------+
// |                                                                           |
// | This program is free software; you can redistribute it and/or             |
// | modify it under the terms of the GNU General Public License               |
// | as published by the Free Software Foundation; either version 2            |
// | of the License, or (at your option) any later version.                    |
// |                                                                           |
// | This program is distributed in the hope that it will be useful,           |
// | but WITHOUT ANY WARRANTY; without even the implied warranty of            |
// | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the             |
// | GNU General Public License for more details.                              |
// |                                                                           |
// | You should have received a copy of the GNU General Public License         |
// | along with this program; if not, write to the Free Software Foundation,   |
// | Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.           |
// |                                                                           |
// +---------------------------------------------------------------------------+
//

if (strpos(strtolower($_SERVER['PHP_SELF']), 'install_defaults.php') !== false) {
    die('This file can not be used on its own!');
}

/*
 * vthemes default settings
 *
 * Initial Installation Defaults used when loading the online configuration
 * records. These settings are only used during the initial installation
 * and not referenced any more once the plugin is installed
 *
 */

/**
* the vthemes plugin's config array
*/
global $_VT_DEFAULT;
$_VT_DEFAULT = array();

/**
 * this lets you select which functions are available for registered users only
 */
$_VT_DEFAULT['vthemesloginrequired'] = 0;

/**
 * Set to 1 to hide the "vthemes" entry from the top menu:
 */
$_VT_DEFAULT['hidevthemesmenu']    = 0;


/*
 * Configuration Options:
 *
 * max_width_preview 	- Sets the maximum width of the thumbnail image
 * colums				- The number of columns to display
 * rows					- The number of rows to display per page
 * disable_menu			- Disable the 'vThemes' link the menu items
 * disable_use_theme	- Disable the 'Use Theme' button
 * disable_download		- Disable the Download Theme link
 *
 */

$_VT_DEFAULT['max_width_preview'] 	= '480';
$_VT_DEFAULT['columns'] 			= 1;
$_VT_DEFAULT['rows']    			= 4;
$_VT_DEFAULT['disable_use_theme'] 	= 0;
$_VT_DEFAULT['disable_download'] 	= 0;

/*
 * Visual Tweaks
 */
 $_VT_DEFAULT['leftblocks']		= '1';

// vthemes Block
$_VT_DEFAULT['block_isleft'] = 0;
$_VT_DEFAULT['block_order'] = 100;
$_VT_DEFAULT['block_topic_option'] = TOPIC_ALL_OPTION;
$_VT_DEFAULT['block_topic'] = array();
$_VT_DEFAULT['block_enable'] = true;
$_VT_DEFAULT['block_permissions'] = array (2, 2, 2, 2);

/**
* Initialize vthemes plugin configuration
*
* Creates the database entries for the configuation if they don't already
* exist. Initial values will be taken from $_VT_CONF if available (e.g. from
* an old config.php), uses $_VT_DEFAULT otherwise.
*
* @return   boolean     true: success; false: an error occurred
*
*/
function plugin_initconfig_vthemes()
{
    global $_VT_CONF, $_VT_DEFAULT, $_GROUPS, $_TABLES;

    if (is_array($_VT_CONF) && (count($_VT_CONF) > 1)) {
        $_VT_DEFAULT = array_merge($_VT_DEFAULT, $_VT_CONF);
    }

    $c = config::get_instance();
    if (!$c->group_exists('vthemes')) {
        $c->add('sg_main', NULL, 'subgroup', 0, 0, NULL, 0, true, 'vthemes', 0);
        $c->add('tab_main', NULL, 'tab', 0, 0, NULL, 0, true, 'vthemes', 0);
        $c->add('fs_access', NULL, 'fieldset', 0, 0, NULL, 0, true, 'vthemes', 0);
        $c->add('vthemesloginrequired', $_VT_DEFAULT['vthemesloginrequired'],
                'select', 0, 0, 0, 10, true, 'vthemes', 0);
        $c->add('hidevthemesmenu', $_VT_DEFAULT['hidevthemesmenu'], 'select',
                0, 1, 0, 20, true, 'vthemes', 0);
		$c->add('fs_visual', NULL, 'fieldset', 0, 1, NULL, 0, true, 'vthemes', 0);
        $c->add('max_width_preview', $_VT_DEFAULT['max_width_preview'], 'text',
                0, 0, 0, 30, true, 'vthemes', 0);
		$c->add('columns', $_VT_DEFAULT['columns'], 'text',
                0, 0, 0, 40, true, 'vthemes', 0);
		$c->add('rows', $_VT_DEFAULT['rows'], 'text',
                0, 0, 0, 50, true, 'vthemes', 0);
		$c->add('disable_use_theme', $_VT_DEFAULT['disable_use_theme'], 'select',
                0, 1, 0, 60, true, 'vthemes', 0);
		$c->add('disable_download', $_VT_DEFAULT['disable_download'], 'select',
                0, 1, 0, 70, true, 'vthemes', 0);
		$c->add('leftblocks', $_VT_DEFAULT['leftblocks'], 'select',
                0, 1, 0, 100, true, 'vthemes', 0);
		
		//Block tab
		$c->add('tab_vthemes_block', NULL, 'tab', 0, 20, NULL, 0, true, 'vthemes', 10);
        $c->add('fs_block_settings', NULL, 'fieldset', 0, 10, NULL, 0, true, 'vthemes', 10);
        $c->add('block_enable', $_VT_DEFAULT['block_enable'], 'select', 
                0, 10, 0, 10, true, 'vthemes', 10);
        $c->add('block_isleft', $_VT_DEFAULT['block_isleft'], 'select', 
                0, 10, 0, 20, true, 'vthemes', 10);
        $c->add('block_order', $_VT_DEFAULT['block_order'], 'text',
                0, 10, 0, 30, true, 'vthemes', 10);
        $c->add('block_topic_option', $_VT_DEFAULT['block_topic_option'],'select',
                0, 10, 15, 40, true, 'vthemes', 10);  
        $c->add('block_topic', $_VT_DEFAULT['block_topic'], '%select',
                0, 10, NULL, 50, true, 'vthemes', 10);
    
        $c->add('fs_block_permissions', NULL, 'fieldset', 0, 20, NULL, 0, true, 'vthemes', 10);
        $new_group_id = 0;
        if (isset($_GROUPS['vthemes Admin'])) {
            $new_group_id = $_GROUPS['vthemes Admin'];
        } else {
            $new_group_id = DB_getItem($_TABLES['groups'], 'grp_id', "grp_name = 'vthemes Admin'");
            if ($new_group_id == 0) {
                if (isset($_GROUPS['Root'])) {
                    $new_group_id = $_GROUPS['Root'];
                } else {
                    $new_group_id = DB_getItem($_TABLES['groups'], 'grp_id', "grp_name = 'Root'");
                }
            }
        }         
        $c->add('block_group_id', $new_group_id,'select',
                0, 20, NULL, 10, TRUE, 'vthemes', 10);        
        $c->add('block_permissions', $_VT_DEFAULT['block_permissions'], '@select', 
                0, 20, 14, 20, true, 'vthemes', 10);
    }

    return true;
}

?>

public_html/index.php
<?php
// +---------------------------------------------------------------------------+
// | Visual Theme Switcher v1 Geeklog Plugin                                   |
// +---------------------------------------------------------------------------+
// | $Id:: index.php 1386 2007-12-04 03:35:03Z mevans0263                     $|
// | index.php                                                                 |
// +---------------------------------------------------------------------------+
// | Copyright (C) 2007 by the following authors:                              |
// |                                                                           |
// | Author: Mark R. Evans - mevans@ecsnet.com                                 |
// +---------------------------------------------------------------------------+
// |                                                                           |
// | This program is free software; you can redistribute it and/or             |
// | modify it under the terms of the GNU General Public License               |
// | as published by the Free Software Foundation; either version 2            |
// | of the License, or (at your option) any later version.                    |
// |                                                                           |
// | This program is distributed in the hope that it will be useful,           |
// | but WITHOUT ANY WARRANTY; without even the implied warranty of            |
// | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the             |
// | GNU General Public License for more details.                              |
// |                                                                           |
// | You should have received a copy of the GNU General Public License         |
// | along with this program; if not, write to the Free Software Foundation,   |
// | Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.           |
// |                                                                           |
// +---------------------------------------------------------------------------+
//
require_once('../lib-common.php');

$T = COM_newTemplate(CTL_plugin_templatePath('vthemes'));
$T->set_file('page','tchooser.thtml');

if (isset($_GET['page'])) {
    $page = COM_applyFilter($_GET['page'], true);
	if ($page < 1) {
		COM_handle404($_CONF['site_url'] . '/vthemes/index.php');
	}
} else {
	// Only time page can be 0 if it is not set
    $page = 0;
}
if ( $page != 0 ) {
    $page = $page - 1;
}

if ( isset($_VT_CONF['override_columns']) ) {
	$columns_per_page   = $_VT_CONF['override_columns'];
} else {
	$columns_per_page   = $_VT_CONF['columns'];
}

if ($columns_per_page == 1) {
    $tdwidth = 'width:100%;';
} else if ($columns_per_page == 2) {
    $tdwidth = 'width:50%;';
} else if ($columns_per_page == 3) {
    $tdwidth = 'width:33%;';
} else if ($columns_per_page == 4) {
    $tdwidth = 'width:25%;';
} else {
    $tdwidth = '';
}

$rows_per_page = $_VT_CONF['rows'];
$items_per_page = $columns_per_page * $rows_per_page;


if ( SEC_inGroup('Root')) {
	$admin = 1;
} else {
	$admin = 0;
}

// Get Current Theme name for page
// *************************************************************
// Once Geeklog 2.2.2 is released replace with COM_getThemeInfo
$theme = vThemes_COM_getThemeInfo($_CONF['theme']);
$tmsg = sprintf($LANG_VT01['active'], $theme['theme_name']);
unset($theme);
$T->set_var('tmsg', $tmsg);

// Retrieve all valid themes
// *************************************************************
// Once Geeklog 2.2.2 is released replace with COM_getThemes
$themes = vThemes_COM_getThemes(false, true, true);

$total_themes = count($themes);
$total_pages = ceil($total_themes/($items_per_page));

if ( $page >= $total_pages ) {
	COM_handle404($_CONF['site_url'] . '/vthemes/index.php');
}

$start = $page * $items_per_page;

$current_print_page = ( floor( $start / $items_per_page ) + 1 );
$total_print_pages  = ceil($total_themes/($items_per_page));

if ( $current_print_page == 0 ) {
    $current_print_page = 1;
}

if ( $total_print_pages == 0 ) {
    $total_print_pages = 1;
}

if ( $total_themes > 0 ) {
    $k = 0;

	$T->set_block('page', 'themeColumn', 'tColumn');
	$T->set_block('page', 'themeRow', 'tRow');
	
	// Have to make sure start on right theme in array for current page so loop through if needed
	for ($i = 0; $i < $start; $i++) {
		$theme = next($themes);
	}
	

    for ( $i = $start; $i < ($start + $items_per_page ); $i += $columns_per_page ) {
        for ($j = $i; $j < ($i + $columns_per_page); $j++) {
            if ($j >= $total_themes) {
                $k = ($i+$columns_per_page) - $j;
                $m = $k % $columns_per_page;
                $T->set_var(array(
                    'themeDisplay'  =>  '',
					'tdwidth'       => $tdwidth,
                ));

                $T->parse('tColumn','themeColumn',true);
                break;
            }
            
			$themeFolder = key($themes);
			$theme = current($themes);
			
			$V = COM_newTemplate(CTL_plugin_templatePath('vthemes'));
			$V->set_file('theme','theme.thtml');

			if (isset($theme['theme_preview_image']) AND !empty($theme['theme_preview_image'])) {
				$previewImage = "/layout/$themeFolder/" . $theme['theme_preview_image'];
			} else {
				$previewImage = '';
			}
			
			// Retrieve theme config options
			$themeDescription = '';
			$themeDescNoHTML  = '';
			$themeHomepage	  = '';
			$themeAuthor	  = '';
			$themeAuthorUrl	  = '';
			$themeDownloadUrl = '';
			$themeVersion	  = '';
			$themeGLVersion	  = '';
			$themeCopyright	  = '';
			$themeLicense	  = '';			
			
				
			$themeName	      = (isset($theme['theme_name']) ? $theme['theme_name'] : $themeFolder);
			
			$themeDescription = (isset($theme['theme_description']) ? $theme['theme_description'] : '');
			if (!empty(trim($themeDescription))) {
				$themeDescNoHTML = preg_match('/<.*>/', $themeDescription); // Will = 0 if no HTML tags
				// Update variable to now indicate if html found or not
				if ($themeDescNoHTML > 0) {
					$themeDescNoHTML = '';
				} else {
					$themeDescNoHTML = 1; 
				}
				if ($themeDescNoHTML) {
					$themeDescription = COM_nl2br($themeDescription);
				}
			}
			
			$themeHomepage	  = (isset($theme['theme_homepage']) ? $theme['theme_homepage'] : '');
			
			$themeAuthor	  = (isset($theme['theme_author']) ? $theme['theme_author'] : '');
			$themeAuthorUrl	  = (isset($theme['theme_author_url']) ? $theme['theme_author_url'] : '');
			
			if ($_VT_CONF['disable_download'] != 1) {
				$themeDownloadUrl 	  = (isset($theme['theme_download_url']) ? $theme['theme_download_url'] : '');
			}
			
			$themeVersion   = (isset($theme['theme_version']) ? $theme['theme_version'] : '');
			$themeGLVersion   = (isset($theme['theme_gl_version']) ? $theme['theme_gl_version'] : '');
			$themeCopyright   = (isset($theme['theme_copyright']) ? $theme['theme_copyright'] : '');
			$themeLicense     = (isset($theme['theme_license']) ? $theme['theme_license'] : '');

			$V->set_var(array(
				'lang_author'           => $LANG_VT00['author'],
				'lang_use_theme'    => $LANG_VT00['use_theme'],
				'lang_theme'		=> $LANG_VT00['theme'],
				'lang_version'		=> $LANG_VT00['version'],
				'lang_requires'		=> $LANG_VT00['requires'],
				'lang_copyright'	=> $LANG_VT00['copyright'],
				'lang_license'		=> $LANG_VT00['license'],
				'lang_description'	=> $LANG_VT00['description'],
				
				'php_self'	    	=> $_SERVER['PHP_SELF'],
				'themeId'	    	=> 'theme' . $i,
				'themeFolder'    	=> $themeFolder,
				'themeTitle'	    => $themeName,
				'themeDescription'  => $themeDescription,
				'themeDescNoHTML'   => $themeDescNoHTML,
				'themeHomepage'		=> $themeHomepage,
				'themeAuthor'		=> $themeAuthor,
				'themeAuthorUrl'	=> $themeAuthorUrl,
				'themeDownloadUrl'	=> $themeDownloadUrl,
				'lang_download' 	=> $LANG_VT00['download_theme'],
				'themeVersion'		=> $themeVersion,
				'themeGLVersion'	=> $themeGLVersion,
				'themeCopyright'	=> $themeCopyright,
				'themeLicense'		=> $themeLicense,
				
				'themeUse'         => ( $_VT_CONF['disable_use_theme'] != 1 ? '1' : '') ,
                'themePreviewImage'   => $previewImage,
				'previewImageWidth' => $_VT_CONF['max_width_preview'], 
			));
			
			$V->parse('output','theme');
			$themeDisplay = $V->finish($V->get_var('output'));

            $T->set_var(array(
                'themeDisplay'  =>  $themeDisplay,
				'tdwidth'       => $tdwidth,
            ));
			$T->parse('tColumn','themeColumn',true);
			
			$theme = next($themes);
		}
        $T->parse('tRow','themeRow',true);
        $T->set_var('tColumn','');
		
		if ($j >= $total_themes) {
			break;
		}
	}
}

$T->set_var(array(
    'bottom_pagination'     => COM_printPageNavigation($_CONF['site_url'] . '/vthemes/index.php', $page+1,ceil($total_themes  / $items_per_page)),
    'title_row'				=> $LANG_VT00['vtheme_header'],
));

$display = $T->finish($T->parse('output','page'));

($_VT_CONF['leftblocks'] == 1) ? $html_infos['what'] = 'menu' : $html_infos['what'] = 'none';

//Output
COM_output(COM_createHTMLDocument($display, $html_infos));
?>


language/english.php
<?php
// +---------------------------------------------------------------------------+
// | Visual Theme Switcher v1 Geeklog Plugin                                   |
// +---------------------------------------------------------------------------+
// | $Id::                                                                    $|
// | This is the English language page for the vThemes Plugin                  |
// +---------------------------------------------------------------------------|
// | Copyright (C) 2002,2005,2006,2007 by the following authors:               |
// |                                                                           |
// | Author:                                                                   |
// | Mark R. Eavns               -    mevans@ecsnet.com                        |
// +---------------------------------------------------------------------------|
// |                                                                           |
// | If you translate this file, please consider uploading a copy at           |
// |    http://www.gllabs.org so others can benefit from your translation.     |
// |    Thanks You!                                                            |
// |                                                                           |
// | This program is free software; you can redistribute it and/or             |
// | modify it under the terms of the GNU General Public License               |
// | as published by the Free Software Foundation; either version 2            |
// | of the License, or (at your option) any later version.                    |
// |                                                                           |
// | This program is distributed in the hope that it will be useful,           |
// | but WITHOUT ANY WARRANTY; without even the implied warranty of            |
// | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the             |
// | GNU General Public License for more details.                              |
// |                                                                           |
// | You should have received a copy of the GNU General Public License         |
// | along with this program; if not, write to the Free Software               |
// | Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA|
// |                                                                           |
// +---------------------------------------------------------------------------|

$LANG_VT00 = array (
    'menulabel'         => 'vThemes',
    'plugin'            => 'vthemes',
    'access_denied'     => 'Access Denied',
    'access_denied_msg' => 'You do not have the proper security privilege to access to this page.  Your user name and IP have been recorded.',
    'install_header'    => 'vThemes Plugin Install/Uninstall',
    'uninstalled'       => 'vThemes is Not Installed',
    'install_success'   => 'vThemes Installation Successful.  <br /><br />Please review the system documentation and also visit the  <a href="%s">administration section</a> to insure your settings correctly match the hosting environment.',
    'uninstall_msg'     => 'Plugin Successfully Uninstalled',
    'install'           => 'Install',
    'enabled'           => 'Disable plugin before uninstalling.',
    'warning'           => 'Warning! Plugin is still Enabled',
    'readme'            => 'vThemes Plugin Installation',
    'overview'          => 'vThemes is a Visual Theme Switcher for Geeklog.<br /><br />',
    'details'           => '',
    'vt_block_header'	=> 'Visual Theme Switcher',
    'preinstall_check'  => 'vThemes has the following requirements:',
    'geeklog_check'     => 'Geeklog v1.4.1 or greater, version reported is <b>%s</b>.',
    'php_check'         => 'PHP v4.3.0 or greater, version reported is <b>%s</b>.',
    'preinstall_confirm' => "For full details on installing vThemes, please refer to the <a href=\"http://www.geeklog.fr/cms/wiki/doku.php/plugins:vthemes\">Installation Manual</a>.",
    'use_theme'			=> 'Use Theme',
    'vtheme_header'		=> 'Visual Theme Switcher',
    'download_theme'	=> 'Download Theme',
    'author'			=> 'Author',
	'theme'				=> 'Theme',
	'version'			=> 'Version',
	'requires'			=> 'Requires',
	'copyright'			=> 'Copyright',
	'license'			=> 'License', 
	'description'		=> 'Description'
);

$LANG_VT01 = array(
	'active'			=> 'Current Active Theme: <em>%s</em>',
	'vt_block'			=> 'Test out available themes by selecting from one of the %d available themes or give the <a href="%s">Visual Switcher</a> a try:<br />',
	'use_permanently'	=> 'Use Permanently'
);

$PLG_vthemes_MESSAGE1 = 'vThemes Plugin successfully installed.';
$PLG_vthemes_MESSAGE2 = 'vThemes Plugin successfully upgraded.';
$PLG_vthemes_MESSAGE3 = 'vThemes Plugin upgrade failed - check error.log';

// Localization of the Admin Configuration UI
$LANG_configsections['vthemes'] = array(
    'label' => 'vthemes',
    'title' => 'vthemes Configuration'
);

$LANG_confignames['vthemes'] = array(
    'vthemesloginrequired' => 'vthemes Login Required?',
    'hidevthemesmenu' => 'Hide vthemes Menu Entry?',
    'max_width_preview' => 'Max width preview?',
    'columns' => 'How many columns?',
	'rows' => 'How many rows?',
	'disable_use_theme' => 'Disable use theme?',
	'disable_download' => 'Disable download?',
	'leftblocks' => 'Leftblocks?',
	'block_enable' => 'Enabled',
    'block_isleft' => 'Display Block on Left',
    'block_order' => 'Block Order',
    'block_topic_option' => 'Topic Options',
    'block_topic' => 'Topic',
    'block_group_id' => 'Group',
    'block_permissions' => 'Permissions'
);

$LANG_configsubgroups['vthemes'] = array(
    'sg_main' => 'Main Settings'
);

$LANG_tab['vthemes'] = array(
    'tab_main' => 'General vthemes Settings',
    'tab_vthemes_block' => 'vthemes Block'
);

$LANG_fs['vthemes'] = array(
    'fs_access' => 'Access vthemes Settings',
    'fs_visual' => 'Visual Settings',
	'fs_block_settings' => 'Block Settings',
    'fs_block_permissions' => 'Block Permissions'
);

// Note: entries 0, 1, and 12 are the same as in $LANG_configselects['Core']
$LANG_configselects['vthemes'] = array(
    0 => array('True' => 1, 'False' => 0),
    1 => array('True' => TRUE, 'False' => FALSE),
    14 => array('No access' => 0, 'Read-Only' => 2,'Read-Write' => 3),
	15 => array('All' => TOPIC_ALL_OPTION, 'Homepage Only' => TOPIC_HOMEONLY_OPTION, 'Select Topics' => TOPIC_SELECTED_OPTION)
);
?>
